### DISCUSSION TITLE
Adding custom key in `kong.conf`

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#8837] [Help]
Hi,

Is it possible to add a custom key in `kong.conf` and use that in `nginx` templating or  plugin code

Thanks in advance

**Comment by jschmid1** [comment]
Could you maybe give a bit more insight in what you're trying to achieve here? A concrete example would really help.

TIA

**Reply by shettyh** [reply]
@jschmid1 Hey, trying to add a new key in kong.conf that can be used in custom `nginx.template`. Like adding to flag `enable_feature_a` and use that to add new section in nginx template if the flag is set

**Reply by hanshuebner** [reply]
@shettyh did you have a look at the [Kong Configuration Reference](https://docs.konghq.com/gateway/latest/reference/configuration/) yet?  It describes how `kong.conf` can contain individual nginx configuration directives as well as how to include bigger nginx configuration blocks.  Or do we misunderstand your request?

Kind regards,
Hans

**Reply by Tieske** [reply]
@bungle we had this discussion a while ago. Don't recall exactly, but sopmething about making `os.getenv` available in the template engine? What did we conclude?

**Reply by Tieske** [reply]
so this one: https://github.com/Kong/kong/pull/6872

So you cannot add it to `kong.conf`, but if you specify it as an environment variable, you can use it in the template. Also in the plugins, but only in the `init_worker` phase. You'd have to read it and cache it.

**Reply by shettyh** [reply]
Ok got it. Thanks

**Reply by Tieske** [reply]
docs have been updated, `os.getenv` is now listed: https://docs.konghq.com/gateway/latest/reference/configuration/#custom-nginx-templates
